.github/workflows/build.yml

```diff
@@ -4,233 +4,187 @@ on:
+          ..
+      working-directory: cmake_build
       env:
-         CMAKE_FLAGS_UBUNTU: -DKEYFINDER=ON -DLOCALECOMPARE=ON -DMAD=ON -DMODPLUG=ON -DWAVPACK=ON
+         PATH: ${{ steps.buildenv.outputs.path }}
+         CMAKE_FLAGS_EXTRA: ${{ matrix.cmake_args }} ${{ steps.apple_codesign.outputs.cmake_args }} -DCMAKE_PREFIX_PATH=${{ steps.buildenv.outputs.cmake_prefix_path }} -DQt5_DIR=${{ steps.buildenv.outputs.qt_path }}
+         MACOSX_DEPLOYMENT_TARGET: ${{ steps.buildenv.outputs.macosx_deployment_target }}
 
-    - name: Set up problem matcher
+    - name: "Set up problem matcher"
       uses: ammaraskar/gcc-problem-matcher@master
 
-    - name: Build
-      run: cmake --build . -j $(nproc)
+    - name: "Build"
+      run: cmake --build .
       working-directory: cmake_build
+      env:
+        CMAKE_BUILD_PARALLEL_LEVEL: 2
+        # GitHub Actions automatically zstd compresses caches
+        CCACHE_NOCOMPRESS: true
+        PATH: ${{ steps.buildenv.outputs.path }}
 
-    - name: Test
-      run: ctest --timeout 45
-      working-directory: cmake_build
+    - name: "Print ccache stats"
+      run: ccache -s
+      env:
+        PATH: ${{ steps.buildenv.outputs.path }}
 
-    - name: Create Debian package
-      run: cpack -G DEB
+    - name: "Test"
+      run: ctest --timeout 45 ${{ matrix.ctest_args }}
       working-directory: cmake_build
-
-    - name: Deploy Debian package
-      uses: actions/upload-artifact@v2
-      with:
-        name: Ubuntu DEB
-        path: cmake_build/*.deb
-
-  build-macos:
-    name: ${{ matrix.os }}
-    runs-on: ${{ matrix.os }}
-    strategy:
-        matrix:
-            os: [macos-10.15]
-    env:
-      APPLE_CODESIGN_IDENTITY: 2C2B5D3EDCE82BA55E22E9A67F16F8D03E390870
-    steps:
-    - name: clone Git repository
-      uses: actions/checkout@v2
-    - name: ccache
-      uses: actions/cache@v2
-      env:
-        cache-name: ccache
-      with:
-        path: /Users/runner/Library/Caches/ccache
-        key: ${{ matrix.os }}-ccache-${{ github.head_ref }}-${{ github.run_number }}
-        restore-keys: |
-          ${{ matrix.os }}-ccache-${{ github.head_ref }}
-          ${{ matrix.os }}-ccache
-    - name: read build environment file
-      id: read_build_environment_file
-      run: |
-        read -r MIXXX_ENVNAME < cmake/macos_build_environment_name
-        echo "::set-output name=buildenv_name::$MIXXX_ENVNAME"
-    - name: macOS build environment cache
-      uses: actions/cache@v2
       env:
-        cache-name: macOS build environment
-      with:
-        path: /Users/runner/buildenv
-        key: macOS-build-environment-${{ steps.read_build_environment_file.outputs.buildenv_name }}
-    - name: import code signing identity
+        QT_QPA_PLATFORM_PLUGIN_PATH: ${{ steps.buildenv.outputs.qt_qpa_platform_plugin_path }}
+        # Render analyzer waveform tests to an offscreen buffer
+        QT_QPA_PLATFORM: offscreen
+        GTEST_COLOR: 1
+        CTEST_PARALLEL_LEVEL: 2
```

Comment:
IIRC there was a reason we had to disable testing in parallel, but it seems this is no longer an issue? Let's keep an eye on this in case tests sporadically fail.

Reply:
I don't rmember exactly, but I think it was before we discovered `QT_QPA_PLUGIN=offscreen`, so maybe the multiple X11 connections caused it?. I always run `ctest -j16` without issues other than the sporadically failing engine tests which also happen on single threaded CI.